File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int num = nums.length / 2;
        
        HashSet<Integer> hashSet = new HashSet<>(Arrays.stream(nums).boxed().collect(Collectors.toList()));
        
        answer = hashSet.size() < num ? hashSet.size() : num;
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = len(nums)/2
    arr = {}
    for num in nums:
        if num in arr:
            arr[num] = arr[num]+1
        else:
            arr[num] = 0
    if len(arr) < len(nums)/2:
        answer = len(arr)
    return answer